src/lighteval/tasks/lighteval_task.py

```diff
@@ -395,12 +395,29 @@ def construct_requests(
                     metric_categories=[MetricCategory.PERPLEXITY],
                 )
             ]
+        if self.has_metric_category[MetricCategory.GENERATIVE_SAMPLING]:
+            # All these tasks require the same generation process - we can do them in one step
```

Comment:
This comment can be removed now or shifted to L421?

Reply:
Hm, no, it's for generative sampling only, to explain why we do `max(self.num_samples)` when sampling (instead of creating one request for 3 samples, one for 5, etc, for ex).

Reply:
What would make it clearer?

Reply:
Actually not sure this comment applies even with your explaination. What do you mean by "all these tasks" ? it used to make sense when we grouped all the GENRATIVE tasks together but not sure now

Reply:
If, for one eval, I use `maj@4` `maj@10` for example - we won't create 2 requests, but one, which will do `maj@10` (so generate 10 samples), and for `maj@4` we'll use the first four

Reply:
So all these tasks would be "If we have several sampling tasks"